go-controller/pkg/config/cni.go

```diff
@@ -71,29 +73,89 @@ func WriteCNIConfig() error {
 
 // ParseNetConf parses config in NAD spec
 func ParseNetConf(bytes []byte) (*ovncnitypes.NetConf, error) {
+	netconf, isRecoverable, err := parseNetConfList(bytes)
+	if err != nil {
+		if isRecoverable {
+			netconf, err = parseNetConfSingle(bytes)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	if netconf.Topology == "" {
+		// NAD of default network
+		netconf.Name = ovntypes.DefaultNetworkName
+	}
+
+	return netconf, nil
+}
+
+func parseNetConfSingle(bytes []byte) (*ovncnitypes.NetConf, error) {
 	netconf := &ovncnitypes.NetConf{MTU: Default.MTU}
 	err := json.Unmarshal(bytes, &netconf)
 	if err != nil {
 		return nil, err
 	}
+
 	// skip non-OVN NAD
 	if netconf.Type != "ovn-k8s-cni-overlay" {
 		return nil, ErrorAttachDefNotOvnManaged
 	}
-	if netconf.Topology == "" {
-		// NAD of default network
-		netconf.Name = ovntypes.DefaultNetworkName
-	} else {
+
+	err = validateNetConfNameFields(netconf)
+	if err != nil {
+		return nil, err
+	}
+
+	return netconf, nil
+}
+
+func parseNetConfList(bytes []byte) (*ovncnitypes.NetConf, bool, error) {
+	confList, err := libcni.ConfListFromBytes(bytes)
+	if err != nil {
+		return nil, true, err
+	}
+
+	if len(confList.Plugins) > 1 {
+		return nil, false, ErrorChainingNotSupported
+	}
+
+	netconf := &ovncnitypes.NetConf{MTU: Default.MTU}
+	if err := json.Unmarshal(confList.Plugins[0].Bytes, netconf); err != nil {
+		return nil, true, err
```

Comment:
shouldn't we return "false" here ? I mean, IIUC, we should only try to parse in "single config" mode if that's the configuration format - meaning, I don't think we should try that because the single configuration in the conflist is not properly formatted.